src/go/serviceconfig/service_config_fetcher.go

```diff
@@ -61,37 +60,61 @@ func NewServiceConfigFetcher(mf *metadata.MetadataFetcher, opts options.ConfigGe
 			Timeout: opts.HttpRequestTimeout,
 		},
 		serviceName: serviceName,
-		mf:          mf,
 		opts:        opts,
 	}
+
+	configIdFetcher := NewConfigIdFetcher(serviceName, opts.ServiceControlURL,
+		scf.client, func() (string, time.Duration, error) { return scf.accessToken() })
+	scf.newConfigId = func() (string, error) {
+		return configIdFetcher.fetchNewConfigId()
+	}
+
+	scf.accessToken = func() (string, time.Duration, error) {
+		if accessTokenFromImds == nil && scf.opts.ServiceAccountKey == "" {
+			return "", 0, fmt.Errorf("If --non_gcp is specified, --service_account_key has to be specified.")
```

Comment:
Unclear how this case is `--non_gcp`, please clarify with comment or try to make the if check clearer.